app/models/miq_request.rb

```diff
@@ -456,6 +458,23 @@ def self.create_request(values, requester, auto_approve = false)
 
     request.post_create(auto_approve)
   end
+  api_relay_class_method(:create_request, :create) do |values, requester, auto_approve|
+    [
+      find_source_id_from_values(values),
+      {
+        :options      => values,
+        :requester    => {"user_name" => requester.userid},
+        :auto_approve => auto_approve
+      }
+    ]
+  end
+
+  def self.find_source_id_from_values(values)
+    MiqRequestMixin.get_option(:src_vm_id, nil, values) ||
+      MiqRequestMixin.get_option(:src_id, nil, values) ||
+      MiqRequestMixin.get_option(:src_ids, nil, values).try(:first)
```

Comment:
Sound right, that's what we do here: [app/models/mixins/process_tasks_mixin.rb#L25](https://github.com/ManageIQ/manageiq/blob/290cac365583a3b279f3e257912092ccd13de047/app/models/mixins/process_tasks_mixin.rb#L25)

In this case we would also need to update the list of IDs for each request to only contain the IDs of the appropriate systems.